src/main/kotlin/uk/gov/dwp/dataworks/encryptionmaterialsprovider/MaterialsResolver.kt

```diff
@@ -0,0 +1,201 @@
+package uk.gov.dwp.dataworks.encryptionmaterialsprovider
+
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.s3.model.EncryptionMaterials
+import com.google.common.cache.Cache
+import com.google.common.cache.CacheBuilder
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import org.apache.hadoop.conf.Configuration
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.net.URI
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.PrivateKey
+import java.security.PublicKey
+import java.security.spec.PKCS8EncodedKeySpec
+import java.security.spec.X509EncodedKeySpec
+import java.time.Duration
+import java.time.LocalDateTime
+import java.util.*
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.spec.GCMParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+/**
+ * Supporting class to [DWEncryptionMaterialsProvider] which handles key creation, retrieval and
+ * encryption / decryption of materials. After initialisation, [getEncryptionMaterials] can be called
+ * with "mode" metadata passed in to perform tasks. See KDoc for method for further info.
+ *
+ * Note that this uses a temporary static public-private [KeyPair] to encrypt and decrypt until such time
+ * as code is written to use the DKS service. See TODO articles in this code for what needs removing.
+ */
+class MaterialsResolver(conf: Configuration, private val s3: AmazonS3, cacheExpirySeconds: Long) {
+
+    //TODO Remove "fs.s3.cse.rsa.public" and "fs.s3.cse.rsa.public" when DKS real code added
+    private val requiredConfiguration = setOf("fs.s3.cse.encr.keypairs.bucket", "fs.s3.cse.rsa.public", "fs.s3.cse.rsa.private")
```

Comment:
We should remove these config items and the related code now that we are using DKS

Reply:
In fact, I think the bulk of this code can now be removed as we can delegate to the `DKSMaterialsProvider` to do the encryption/decryption?

Reply:
I retained on purpose and had a parallel implementation for dks so that this can be leveraged for any spikes if need to

Reply:
Ok, I think we should add the `@Deprecated` tag to this then and explain it is superceded by the code in the other package?